### PR TITLE
Fixes #11289: Missing documentation on Directive ordering

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/11289

### CODE REVIEW

File: 4_advanced_usage/35_directive_ordering.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+=== Directives ordering
+
```
Comment (amousset) [inline 4_advanced_usage/35_directive_ordering.txt]: Could you add a paragraph about the motivations behind this (avoid having to wait for multiple runs for convergence) and what should be avoided (a configuration that relies on the order to be applicable)?

**Comment by ncharles** [comment]
Commit modified

File: 4_advanced_usage/35_directive_ordering.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Configurations on Rudder are based on desired states, describing the expected state of the system. However, there are cases where having order is desirable (like ensuring that a JVM is present before deploying an Application server, or ensuring a user is present before setting it sudoers), even if it will converge over the course of several agent runs.
+
+When policies are generated for each nodes, all Directives based on a same Technique are merged together, preventing a complete ordering of all Directives. However, Rudder uses a best-effort method for ordering Directives, based on alphanumeric ordering.
+
+Rudder tries to order first on Rules name, then on Directives name.
+
+Example:
+- given three Techniques A, B and C
```
Comment (amousset) [inline 4_advanced_usage/35_directive_ordering.txt]: We need a blank line here.

File: 4_advanced_usage/35_directive_ordering.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+=== Directives ordering
+
+Configurations on Rudder are based on desired states, describing the expected state of the system. However, there are cases where having order is desirable (like ensuring that a JVM is present before deploying an Application server, or ensuring a user is present before setting it sudoers), even if it will converge over the course of several agent runs.
```
Comment (amousset) [inline 4_advanced_usage/35_directive_ordering.txt]: Configuration in Rudder

**Comment by ncharles** [comment]
Commit modified

File: 4_advanced_usage/35_directive_ordering.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Rudder tries to order first on Rules name, then on Directives name.
+
+Example:
+
+- given three Techniques A, B and C
+- directives A1 and A2 based on Technique A, directives B1 and B2 based on B, directives C1 and C2 based on C
+- rule R1 having A1, B1 and C2 and rule R2 having A2, B2 and C2, both applied on a same node,
+- resulting ordering of directive will be: A1, A2, B1, B2, C2, C1
+
+Please note that you should avoid configuration that relies on order to be applicable, due to the lack of strict ordering for now.
```
Comment (amousset) [inline 4_advanced_usage/35_directive_ordering.txt]: missing empty line at the end of the file

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**